src/Common/src/TypeSystem/Common/CastingHelper.cs

```diff
@@ -354,7 +359,7 @@ private static bool CanCastToClass(this TypeDesc thisType, TypeDesc otherType, S
 
                 if (curType.IsInterface)
```

Comment:
Would it make sense to drop this check completely so that the variant and non-variant paths are more similar?

If curType is interface, BaseType will be null and we will naturally return false below.

Reply:
Yes, that is true. I was not sure whether I should drop it, since it will cause an "unnecessary" while-cycle, but I also prefer the version without it to be honest.